### DISCUSSION TITLE
[ideas] Check overall dependencies of dynamic targets before checking individual dependencies

### DISCUSSION THREAD

**Discussion by kendonB** [ropensci/targets#1148] [Ideas]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I have a project that takes a wee bit of time to get through checking a dynamic target with a few thousand chunks.

I think targets could get a decent speed up in this case if there was a first round of dependency checking that looked at all the inputs together before examining the individual dependencies of each chunk.

A tradeoff of course, but I think almost all of the time a dynamic target will either all be out of date or all up to date. This will save time for the latter and cost a very small amount of extra time for the former.

**Comment by wlandau** [comment]
Unfortunately this is not feasible because of how `targets` tracks dependencies. Each branch is treated like any other target, with a single hash in `tar_meta()$depend` which changes if any of its dependencies change. To unpack `tar_meta()$depend` into individual hashes would not be feasible, and it would make most things run much slower.

The last couple versions are much faster at churning through larger numbers of targets, and I added a few things to https://books.ropensci.org/targets/performance.html, including `seconds_meta_append`, `seconds_reporter`, and `trust_object_timestamps = TRUE`. Beyond that, you could batch the same amount of work in fewer targets, or share an example pipeline that I could profile.
